api/provider/aws/task_list.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+const MAX_TASK_IDS = 100
+
+func (t *TaskProvider) List() ([]models.TaskSummary, error) {
+	clusterNames, err := t.listClusterNames()
+	if err != nil {
+		return nil, err
+	}
+
+	taskCopies := map[string][]*ecs.Task{}
+	for _, clusterName := range clusterNames {
+		environmentID := delLayer0Prefix(t.Config.Instance(), clusterName)
+
+		running := "RUNNING"
+		tasks, err := t.listTasksByStatus(&running, &environmentID)
+		if err != nil {
+			return nil, err
+		}
+
+		stopped := "STOPPED"
+		stoppedTasks, err := t.listTasksByStatus(&stopped, &environmentID)
+		if err != nil {
+			return nil, err
+		}
+
+		taskARNs := append(tasks, stoppedTasks...)
+		if len(taskARNs) > 0 {
```

Comment:
Couple things:

* You're putting aws api calls into the "master" function - this should be placed in a helper function
* I understand this is copied over from the previous implementation. However, it's a bad implementation that needs to be re-analyzed with a critical eye. A huge red flag is we are doing a `Describe` call during a `List`. We should avoid that as much as possible. 
* The notion of `taskCopies` can go away, as it was carryover from old implementations of tasks. We have all the info we need in the `listTasksByStatus`. We shouldn't need to make any more api calls after that.